### ISSUE TITLE
Feature: Offline ID assignment and online synchronization

### ISSUE THREAD

**Issue by vjekob** [vjekob/al-objid#38]
Ninja should be able to assign new object IDs when offline. Intellisense should indicate offline suggestions with a different icon/color, and it should be able to detect when it goes online. After it goes online, it should sync the offline consumption, and should be able to detect if any of offline IDs are still available, and if not, it should re-assign (re-number) those that are not.

**Comment by vjekob** [comment]
This is being addressed in the upcoming **3.4.1** release, and this issue will be closed with it.

Rather than full offline ID assignment with automatic post-reconnect renumbering — which would reintroduce the very cross-developer collision risk Ninja exists to prevent — 3.4.1 ships the high-value "offline grace" improvements:

- **Reliable offline detection.** Ninja now correctly recognizes when the back end is unreachable across the common failure modes (`ENOTFOUND`, `ENETUNREACH`, `ECONNREFUSED`, `ETIMEDOUT`). Previously most of these went undetected.
- **No more notification spam.** The "cannot communicate with the back end" message now appears only once, on the transition from online to offline, instead of on every failed request — and it clears automatically once connectivity is restored.

Automatic offline allocation and renumbering remain intentionally out of scope; the existing **Store Assignment** and **Synchronize Object IDs** commands cover the manual reconciliation workflow once you're back online.

**Comment by vjekob** [comment]
On reflection, I'm going to close this for the broader offline-assignment scope as **won't fix**.

The way object IDs get assigned has shifted decisively toward agent-driven workflows, and in that model a full offline mode no longer makes sense to build or maintain — the added complexity and the collision-risk trade-offs outweigh the benefit, especially since assignment is no longer a manual, connection-bound step. This request has also been open for four years with no further interest beyond the original ask, so keeping it open isn't serving anyone.

The "offline grace" improvements in 3.4.1 — reliable offline detection and a single, non-repetitive notification that clears automatically on reconnect — already cover the real-world pain of being briefly disconnected. For anything beyond that, the existing **Store Assignment** and **Synchronize Object IDs** commands handle reconciliation once you're back online.

Thanks for the original request!
